projects/indicators/src/modules/text-highlight/text-highlight.directive.ts

```diff
@@ -112,50 +127,45 @@ export class SkyTextHighlightDirective
     return regex.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
   }
 
-  private static getRegexMatch(
-    node: HTMLElement,
-    searchText: string
-  ): RegExpExecArray {
-    const text = node.nodeValue;
-    const newSearchText = this.cleanRegex(searchText);
-    const searchRegex = new RegExp(newSearchText, 'gi');
-
-    return searchRegex.exec(text);
-  }
-
-  private static markNode(node: any, searchText: string): number {
-    const regexMatch = SkyTextHighlightDirective.getRegexMatch(
-      node,
-      searchText
-    );
+  private static markNode(node: any, searchTerms: string[]): number {
+    /* istanbul ignore else */
+    if (searchTerms) {
+      const text = node.nodeValue;
+      for (let index = 0; index < searchTerms.length; index++) {
+        searchTerms[index] = this.cleanRegex(searchTerms[index]);
+      }
+      const searchRegex = new RegExp(searchTerms.join('|'), 'gi');
 
-    // found match
-    if (regexMatch && regexMatch.length > 0) {
-      // split apart text node with mark tags in the middle on the search term
-      const matchIndex = regexMatch.index;
+      let match;
+      while ((match = searchRegex.exec(text))) {
```

Comment:
Why is this using `while` if it doesn't loop?

Reply:
You can use the `.exec()` method multiple times to find successive matches in the same string. When it's used in a `while` loop like this, It keeps evaluating the string for the match, starting with the substring of the index of the last found item. You can see an example here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec#examples

Reply:
On deeper review, the while loop is actually being called by another recursive loop inside the `markTextNodes()` method, so it was hiding the fact that the while loop returns `1` each time. Since the regex logic is already wrapped inside this recursion, I just removed the `while` loop altogether. Good catch.